configure

```diff
@@ -68,42 +70,100 @@ while true; do
       ;;
 
     --readline=*)
-      # \( is required for non-GNU expr, which doesn't like leading dashes
-      # See https://www.freebsd.org/cgi/man.cgi?query=expr&sektion=&n=1,
-      # specifically the section on 'Compatibility with previous implementations'
-      FLAG_readline=$(expr \( "$1" : '--readline=\(.*\)' \) )
+      FLAG_readline="${1#*=}"
       ;;
     --readline)
       if test $# -eq 1; then
         die "--readline requires an argument"
       fi
       shift
-      FLAG_readline=$1
+      FLAG_readline="$1"
       ;;
 
     # TODO: Maybe prefix only needs to be part of the install step?  I'm not
     # sure if we need it for building anything.
     --prefix=*)
-      FLAG_prefix=$(expr \( "$1" : '--prefix=\(.*\)' \) )
+      FLAG_prefix="${1#*=}"
       ;;
     --prefix)
       if test $# -eq 1; then
         die "--prefix requires an argument"
       fi
       shift
-      FLAG_prefix=$1
+      FLAG_prefix="$1"
       ;;
 
     # Following autoconf's spelling of --mandir
     --datarootdir=*)
-      FLAG_datarootdir=$(expr \( "$1" : '--datarootdir=\(.*\)' \) )
+      FLAG_datarootdir="${1#*=}"
       ;;
     --datarootdir)
       if test $# -eq 1; then
         die "--datarootdir requires an argument"
       fi
       shift
-      FLAG_datarootdir=$1
+      FLAG_datarootdir="$1"
+      ;;
+
+    --with-*|--enable-*)
+      info "Argument ignored '$1'"
+      ;;
+
+    --build=*)
```

Comment:
Since we're not using $1, please collapse these flags for readability:
```
--build|--build=*)
  info "Ignored argument '$1'"
  ;;
```

Or I think you can even do all of them in a single column while still keeping the 80 char limit.

Reply:
That will be a bug if someone does `--ignored notignored`

Reply:
Is there an argument against using `getopt` (or was it `getopts`? I keep forgetting which one it was)

Reply:
OK I see, well you could have exactly two cases then:

1. --build, --host, etc.
2. --build=, --host=, etc.

There is a POSIX builtin getopt, but it doesn't take long flags, only short flags.

Reply:
To be honest I'm not 100% sure we need the `--arg opt` cases anyways. I'm currently trying to figure out if that is standard for configure scripts at all.
After all we're targeting automated build-processes with this change and they can be expected to produce standard-conforming input.